### PR TITLE
Generate doc

### PR DESCRIPTION
**Pull request by ailinvenerus**
**Done:**
- Create docstring
- Update requirements.txt
- Create pipeline to publish to docs repo

### CODE REVIEW

File: .github/workflows/update-doc.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/checkout@v4
+      - name: Install deps
+        run: pip install -r requirements.txt
+      - name: Create md files
+        run: sphinx-build -M markdown ./docs_source ./docs
+      - name: Update docs website
+        uses: car-on-sale/action-pull-request-another-repo@v1.3.1
+        env:
+          API_TOKEN_GITHUB: ${{ secrets.WEBSITE_GITHUB_TOKEN }}
+          GH_TOKEN: ${{ secrets.WEBSITE_GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (ailinvenerus) [reply .github/workflows/update-doc.yml:22]: It's the same token we used in ScribeAuth to update the doc https://github.com/ScribeLabsAI/ScribeAuth/blob/master/.github/workflows/update-doc.yml

**Comment by EHadoux** [comment]
> > lgtm provided Alex's issues are resolved. Ideally we could also see the generated .md with the PR.
> 
> I added the _docs_ folder (with the .md) in .gitignore, should I delete that line and add the files?

Not sure, that might mess up the build/deployment. Just drop it in the PR in a message as an attachment

**Comment by ailinvenerus** [comment]
The .md created when generating the doc:
[reference.md](https://github.com/ScribeLabsAI/ScribeMi/files/13600443/reference.md)

**Comment by ailinvenerus** [comment]
new version: [reference.md](https://github.com/ScribeLabsAI/ScribeMi/files/13603170/reference.md)

File: scribemi/ScribeMi.py [hunk 2]
```diff
@@ -40,6 +94,80 @@ class SubmitTaskParams(SubmitTaskParamsBase, total=False):
...
+    tag: str
+    term: str
+    ogterm: str
+    values: List[Value]
+
+
+class Table(TypedDict):
+    title: str
+    columnsOrder: List[str]
+    items: List[Item]
```
Comment (EHadoux) [inline scribemi/ScribeMi.py:112]: Are we not writing documentation for these?

**Review by EHadoux — CHANGES_REQUESTED** [review]

**Review by alexbostock — COMMENTED** [review]
I'm happy with everything once Emmanuel's comments are resolved.

Some nitpicks about words, but we have another ticket for that: makes sense to nitpick all the docs together as part of the other ticket

**Review by EHadoux — APPROVED** [review]

**Pull request merged**